api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -27,16 +39,100 @@ public NameSupport() {
 	}
 	
 	public static NameSupport getInstance() {
-		if (singleton == null) {
-			throw new APIException("Not Yet Instantiated");
-		} else {
-			return singleton;
+		synchronized (NameSupport.class) {
+			if (singleton == null) {
+				singleton = new NameSupport();
+			}
+		}
+		singleton.init();
+		return singleton;
+	}
+
+	private void init() {
+		if (!initialized) {
+			try {
+				Context.getAdministrationService().addGlobalPropertyListener(singleton);
+				// Get layout NameTemplate format to override the default if any
+				String layoutTemplateXml = Context.getAdministrationService().getGlobalPropertyObject(
+					OpenmrsConstants.GLOBAL_PROPERTY_LAYOUT_NAME_FORMAT).toString();
+				setNameTemplate(layoutTemplateXml);
+			} catch (NullPointerException ignored) {}
+
+			List<String> specialTokens = new ArrayList<>();
+			specialTokens.add("prefix");
+			specialTokens.add("givenName");
+			specialTokens.add("middleName");
+			specialTokens.add("familyNamePrefix");
+			specialTokens.add("familyName");
+			specialTokens.add("familyName2");
+			specialTokens.add("familyNameSuffix");
+			specialTokens.add("degree");
+			setSpecialTokens(specialTokens);
+			initialized = true;
+		}
+	}
+	
+	public void setNameTemplate(List<NameTemplate> nameTemplates) {
+		this.layoutTemplates = nameTemplates;
+		setDefaultLayoutFormat(nameTemplates.get(0).getCodeName());
```

Comment:
Why the call to `setDefaultLayoutFormat()`? Shouldn't that be separately controlled by the `layout.name.format` GP?